src/components/Calculator.js

```diff
@@ -1,68 +1,86 @@
-import React from 'react';
+import { React, useState } from 'react';
 import styles from './Calculator.module.css';
+import calculate from '../logic/calculate';
 
 export default function Calculator() {
+  const [total, setTotal] = useState(0);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+
+  function clickHandler(e) {
+    const result = calculate({ total, next, operation }, e.target.textContent);
+    setTotal(result.total);
+    setNext(result.next);
+    setOperation(result.operation);
+  }
+
```

Comment:
- [ ] Nice work done on this project. Your Calculator component should be a class-based component for now. There is nothing wrong with how you have it now, but the project requirements state that you should use _**React Lifecycle methods**._ The only way you can use a lifecycle method is by turning your component into a class-based component. You can read this [article](https://www.w3schools.com/react/react_lifecycle.asp) to have an idea of how to use lifecycle methods. 👍

Reply:
I completely missed that requirement, I just did the refactoring, thanks! 😁